[UI] Add dark theme generation for Clarity components

hippee-lee opened the pull request:
## Demo
http://clarity-ks-dark.surge.sh/alerts

## Build Changes
`webpack.config.js` has been modified to consume the new `clarity-dark.scss` file and output the dark theme css file for both production and test. 

## SCSS Changes
These are the most comprehensive changes. I think *every* scss file has been affected. Some files have more significant changes than others. The main changes were to support a centralization of scss variables. 

### Component level changes
SCSS variables have been abstracted out of the `_component.clarity.scss` file and placed into a `_variables.component.scss` file. At this point, there should not be any $clr-variables in the `_component.clarity.scss` file.

### main.scss & clarity-dark.scss
Both of these are top level / entry points for the light and dark themes. `main.scss` has changes to two simple imports

1. utils/dependencies.clarity.scss
2. utils/components.clarity.scss

Additionally when you look into the `./utils/dependencies.clarity.scss` you will notice that all of the `_variables.component.scss` files have been aggregated into the `utils/variables.clarity.scss` file. and included before the second batch of Bootstrap 4 dependencies. This is how I preserved the Light theme and simplified the top level scss file so that we could use that to build dark theme. 

In the `dark-theme.scss`, sandwiched between the dependencies and components is the `utils/theme.dark.clarity.scss` file where I set the variables to the dark value needed for the dark theme output. 

### utils/theme.dark.clarity.scss
This file is organized by component and where it makes sense I referenced where in our `_component.clarity.scss` codebase the variables are used. Additionally for things that are tricky or more involved to theme I have left some line comments that should be helpful in reminding what part of the component the variable affects. 

## Notes
- There may be a better way to organize the theme file with the `!default` syntax but I held off on further restructuring when I read the docs on how that works. WHat I have now seems inverse to what it could be but I'm not 100% on that. 
- There should be no more $scss-variables in the `_component.clarity.scss` files so If I missed something please let me know. 
- the dev experience is sub-optimal at the moment. First, because of our current webpack setup both theme files are built and output in the same config which means they are also both added to the demo app when we run webpack-dev-server. So in order to run `npm cssTest set1` I have been commenting out `utils/theme.dark.clarity`. Additionally this line needs to be commented out when you push to a branch that will be tested on Travis. 
- I updated a couple cssReference image for a few different reasons
- I still have one more dark theme addition to add for Wizard Ghost Pages

## Preserving this from the git commit
- Centralizes scss variables so that they can be changed by theme developer
- Add Dark Theme values and generate second css file for dark theme
- Organize / document dark theme overwrites as an example
- Preserve light theme generation with minimal changes to cssTest reference images.
- closes #968

## Filesizes
After theme generation I took a screengrab of both the light and dark theme files generated.
Both Clarity light and dark theme files are the same at 449K. 
<img width="1134" alt="screen shot 2017-11-06 at 2 15 10 pm" src="https://user-images.githubusercontent.com/433692/32467003-ddceca98-c2fd-11e7-863f-3e8d376c810e.png">


Signed-off-by: Matt Hippely <mhippely@vmware.com>

lil-kim commented:
Hi @hippee-lee,
Sorry, I forgot to provide button and text link styles for alerts. 
![image](https://user-images.githubusercontent.com/23196444/32470176-3128e5c2-c30b-11e7-81e0-18d03947af74.png)
These "FIX" buttons should be black outline style. Like this:
![image](https://user-images.githubusercontent.com/23196444/32470264-af12a810-c30b-11e7-8d6e-dad7cc7312e1.png)
Here's the spec:
[Buttons spec for color bgs.zip](https://github.com/vmware/clarity/files/1448169/Buttons.spec.for.color.bgs.zip)

lil-kim commented:
Disabled icon buttons need to be updated with dark theme colors:
Should have 100% opacity
Border: #6D7884 
Icon: #8F9BA3
![image](https://user-images.githubusercontent.com/23196444/32470345-26674b96-c30c-11e7-951a-1f6d20317fa8.png)

lil-kim commented:
Text link colors need to be updated to dark theme version.
![image](https://user-images.githubusercontent.com/23196444/32470528-357c6ed0-c30d-11e7-9402-894abe09a009.png)
[Buttons spec for color bgs.zip](https://github.com/vmware/clarity/files/1448169/Buttons.spec.for.color.bgs.zip)

hippee-lee commented:
@lil-kim - I addressed the three issues you mentioned above and republished it to http://clarity-ks-dark.surge.sh/alerts

gnomeontherun commented:
Is the dropdown icon right here? Its black.
![screen shot 2017-11-10 at 9 31 26 am](https://user-images.githubusercontent.com/79500/32665465-f0d6e068-c5f9-11e7-877c-eb64a8c7ac81.png)

Is there an inverted spinner for dark?

hippee-lee commented:
I updated the dropdown icnon and header color spotted in Jeremy's feedback.

adityarb88 commented on package.json:
```diff
@@ -3,7 +3,7 @@
     "version": "0.10.12",
     "private": true,
     "scripts": {
-        "start": "webpack-dev-server --port=4200 --host 0.0.0.0 --open",
```
But I thought the host flag was needed to access the demo app on VMs. Why have we removed that? Also what does the `-d` flag do?

adityarb88 commented on package.json:
```diff
@@ -23,7 +23,7 @@
         "tslint:check": "node_modules/.bin/tslint -c tslint.json \"src/**/*.ts\" -e \"**/node_modules/**\"",
         "tslint:fix": "node_modules/.bin/tslint -c tslint.json --fix \"src/**/*.ts\" -e \"**/node_modules/**\"",
-        "webpack": "webpack",
+        "webpack": "webpack --progress --colors",
```
What do you think about this? https://github.com/FormidableLabs/webpack-dashboard Not for this PR but wanted to note here so that you get a chance to take a look at that.

hippee-lee replied:
Noted, I put it on the list for dev happiness here: https://github.com/vmware/clarity/issues/1624

adityarb88 commented on src/app/login/login.demo.html:
```diff
@@ -28,8 +28,8 @@ <h5 class="hint">Use your Company ID to sign in or create one now</h5>
                 <div class="error active">
-                    Invalid user name or password
-                </div>
+                                Invalid user name or password
```
Formatting change?

hippee-lee replied:
Might be a rebase change. I will un-indent the lines.

adityarb88 commented on src/app/nav/header-types.demo.html:
```diff
@@ -6,7 +6,7 @@
 <div class="clr-example">
     <div class="main-container">
-        <header class="header-6 space-below">
+        <header class="header-1 space-below">
```
This file was for displaying the different sections in the headers. We kept the color same so that we could focus on the sections. There is a separate demo for header colors. Up to you if you want to revert this back but wanted to let you know what this file intended to do.

hippee-lee replied:
I saw the `header-colors` demo later on and meant to put this one back. 
I'll revert it.

adityarb88 commented on src/clarity-angular/button/_toggles.clarity.scss:
```diff
@@ -27,7 +12,7 @@ $toggle-transition-animation: ease-in;
         display: inline-block;
-        height: $toggle-height;
+        height: $clr-toggle-height;
         margin-right: 1rem;
```
Feels a bit weird that height has a variable assigned to it but margins and borders use values directly here. Would it make more sense to keep the variables in one file or just move the color variables in the variables file and use the size ones directly in the CSS? Otherwise we would have to jump around different files to figure out whats going on, no? Let me know what you think about this.

adityarb88 commented on src/clarity-angular/button/_toggles.clarity.scss:
```diff
@@ -36,7 +21,7 @@ $toggle-transition-animation: ease-in;
             cursor: pointer;
-            height: $toggle-height;
+            height: $clr-toggle-height;
             margin-right: 2rem;
```
Same feedback about variables in different files applies here.

adityarb88 commented on src/clarity-angular/button/_toggles.clarity.scss:
```diff
@@ -36,7 +21,7 @@ $toggle-transition-animation: ease-in;
             position: relative;
             cursor: pointer;
-            height: $toggle-height;
+            height: $clr-toggle-height;
```
This variable was not named with the `clr-` prefix because it was local to the file and was used for calculations to compute other values based on the height of the toggle. What we planned on doing was to prefix global variables but not the local variables. But not since we have different files that changes. Just a comment. No action needed but you can let me know what you think about this.

adityarb88 commented on src/clarity-angular/data/_tables.clarity.scss:
```diff
@@ -2,39 +2,15 @@
-        color: clr-getTextColor($clr-table-bgcolor, text);
+        color: $clr-table-font-color;
         margin: 0;
         margin-top: 1rem;
```
Yes I am definitely not sure about variables keeping half variables in the file and half outside. May be we should just move size variables in the file and keep the color variables outside? This can be addressed later on but we need to decide what to do.

adityarb88 commented on src/clarity-angular/data/datagrid/_variables.datagrid.scss:
```diff
@@ -0,0 +1,20 @@
+$clr-datagrid-popover-border-color: $clr-light-midtone-gray;
+$clr-datagrid-action-popover-hover-color: $gray-light;
+
+// TODO - Remove? No usages found
```
If you cant find the usage then yes probably remove it.

adityarb88 commented on src/clarity-angular/data/stack-view/_stack-view.clarity.scss:
```diff
@@ -81,19 +79,19 @@ $clr-stack-font-weight: clr-getTypePropertyValueForDomElement(stackview_text, fo
                 width: 0;
                 height: 0;
-                border-top: 0.375rem solid $clr-blue;
+                border-top: 0.375rem solid $clr-stack-bloc-changed-border-top-color;
```
Is this named incorrectly? should it be`$clr-stack-block-.....`

hippee-lee replied:
Correct. Will fix the typo.

adityarb88 commented on src/clarity-angular/data/stack-view/_stack-view.clarity.scss:
```diff
@@ -197,36 +196,6 @@ $clr-stack-font-weight: clr-getTypePropertyValueForDomElement(stackview_text, fo
             height: 1rem;
         }
 
-        .stack-block-expandable > .stack-block-content {
```
Why is this removed?

hippee-lee replied:
Will mark this for cleanup instead of removing.

adityarb88 commented on src/clarity-angular/data/tree-view/_variables.tree-view.scss:
```diff
@@ -0,0 +1,31 @@
+// TODO - Remove? No usages found
+$clr-tree-node-children-margin-big: $clr-tree-node-touch-target-big - $clr-tree-caret-size / 2 - $clr-rem-1px;
+$clr-tree-node-children-margin-small: $clr-tree-node-touch-target-small - $clr-tree-caret-size / 2 - $clr-rem-1px;
+$clr-tree-node-caret-hover-color: $clr-black;
```
Most of these repeated variables already are present above with a different name. We can remove these.

adityarb88 commented on src/clarity-angular/emphasis/alert/_alert.clarity.scss:
```diff
@@ -126,11 +64,11 @@ $clr-alert-colors: (
     .alert-item > span,
     .alert-text {
-        display: inline-block;
+        //display: inline-block; // TODO get buy in on removal during PR
```
This is needed for IE11. Why are we commenting this out?

hippee-lee replied:
I thought it only flex-basis was required. Will uncomment.

hippee-lee replied:
Added a comment for the attrs req'ed for IE11 and uncommented this.

adityarb88 commented on src/clarity-angular/emphasis/alert/_alert.clarity.scss:
```diff
@@ -126,11 +64,11 @@ $clr-alert-colors: (
         flex-shrink: 1;
         flex-basis: 98%; //needed for IE11
-        max-width: 98%;
+        //max-width: 98%; // TODO get buy in on removal during PR
```
This is needed for IE11. Why are we commenting this out?

hippee-lee replied:
I thought it only flex-basis was required. Will uncomment.

hippee-lee replied:
Added a comment for the attrs req'ed for IE11 and uncommented this.

adityarb88 commented on src/clarity-angular/emphasis/alert/_alert.clarity.scss:
```diff
@@ -282,6 +220,7 @@ $clr-alert-colors: (
                 @extend .btn-inverse;
                 @extend .btn-sm;
                 margin: 0;
+                //background: transparent;
```
Can we remove this comment?

hippee-lee replied:
Will remove.

adityarb88 commented on src/clarity-angular/forms/_inputs.clarity.scss:
```diff
@@ -6,10 +6,11 @@
     #{$styledInputs} {
         @include custom-inputs(1rem);
         @include clr-form-fields-appearance(2.5rem, $clr-form-field-border-color);
+        //color: $clr-input-text-color;
```
Can we remove this comment?

hippee-lee replied:
Will remove.

hippee-lee replied:
Will remove.

adityarb88 commented on src/clarity-angular/forms/_variables.inputs.scss:
```diff
@@ -0,0 +1,21 @@
+
+
+// TODO - Remove? No usages found
+$clr-input-placeholder-text-color: $clr-black;
```
Yes

adityarb88 commented on src/clarity-angular/forms/_variables.select.scss:
```diff
@@ -0,0 +1,13 @@
+ */
+
+// Usage: ../forms/_select.clarity.scss
+$select-hover-background: rgba(221, 221, 221, 0.5);
```
Should we prefix these variables with `clr`?

hippee-lee replied:
yes. Will update.

hippee-lee replied:
Changed the name's

adityarb88 commented on src/clarity-angular/layout/_card.clarity.scss:
```diff
@@ -120,173 +113,173 @@ $card-children-space-between: 0.5rem !default;
-    .card-img {
-        &:first-child > img {
-            border-radius: $clr-default-borderradius $clr-default-borderradius 0 0;
+        .card-text {
```
These look like formatting changes. Are they needed?

hippee-lee replied:
I'll fix these.

hippee-lee replied:
This was introduced when rebasing. I will fix the indentation.

adityarb88 commented on src/clarity-angular/layout/main-container/_variables.header.scss:
```diff
@@ -0,0 +1,34 @@
+$clr-header-7-bg-color: #314351;
+
+// TODO - Remove? No usages found
+$clr-header-dropdown-caret-distance: 0.0833333rem;
```
Yes

adityarb88 commented on src/clarity-angular/popover/signpost/_signposts.clarity.scss:
```diff
@@ -83,36 +87,36 @@ $clr-signpost-innerBlock: $clr-white !default;
          $clr-signpost-pointer-invisible-border: $clr-signpost-border-size solid transparent;
          $clr-signpost-pointer-psuedo-border: $clr-signpost-border-size solid $clr-signpost-innerBlock;
- 
+
```
Formatting changes. If needed to be done they should be in a separate PR.

hippee-lee replied:
I removed some variables from here (I don't think its reflected in this commit) and left one line between the comment and the start of the `&.top-left` rule.

adityarb88 commented on webpack.config.js:
```diff
@@ -391,6 +441,8 @@ module.exports = {
   "devServer": {
     "historyApiFallback": true,
+    "port": 4200,
+    "host":"0.0.0.0",
```
noticed that you added the params here. Ignore the comment i posted earlier :D

lil-kim commented:
@hippee-lee I forgot to provide specs for disabled-checked checkbox. Please update with the hex #s below:
![image](https://user-images.githubusercontent.com/23196444/32746248-d947d732-c869-11e7-99aa-698a5e4fb46d.png)

lil-kim commented:
Status labels shouldn't be clickable (according to light theme documentation).
![image](https://user-images.githubusercontent.com/23196444/32752339-aae12470-c87d-11e7-8ddf-974c796d25b1.png)

lil-kim commented:
@hippee-lee, My bad...you and I discussed lightening the bg on signposts since it floats over other elements.
![image](https://user-images.githubusercontent.com/23196444/32753727-d2e2bbe6-c882-11e7-82e4-b10bd1da2054.png)

lil-kim commented:
Change background overlay color:
![image](https://user-images.githubusercontent.com/23196444/32754474-aaa4eed0-c885-11e7-812e-0257a7798711.png)

lil-kim commented:
**Spinners and Progress bars:**
Please adjust colors to:
![image](https://user-images.githubusercontent.com/23196444/32756804-d6f19f78-c890-11e7-9788-015786c78bbd.png)
Also, pls update the red and green progress bar colors to dark theme danger & success colors:
![image](https://user-images.githubusercontent.com/23196444/32756981-d7806676-c891-11e7-8ddb-ab188db0bfcb.png)

red: #F54F47
green: #60B515

gnomeontherun commented on src/clarity-angular/utils/_maps.clarity.scss:
```diff
@@ -0,0 +1,590 @@
+//Element Properties Map
+$clr-elements: (
+    h1: $clr-h1,
+    h1-dark: #clr-h1-dark,
```
You mean `$clr-h1-dark`?

hippee-lee replied:
Nice catch, this isn't needed anymore. I will remove.

gnomeontherun commented on src/clarity-angular/utils/_theme.dark.clarity.scss:
```diff
@@ -0,0 +1,694 @@
+ * Description the color for links and the active, hovered and visited state
+ * Usages: ./_reboot.clarity.scss
+ */
+$clr-link-active-color: #49afd9;
```
Some of these colors are reused in the file, should we hoist them up into local variables?

hippee-lee replied:
For now I would like to keep them as is b/c we are still tweaking the design a bit. WHen I first started I made variables that tried to semantically group the color usage into a single variable but there has been some churn that has made it difficult to maintain.

gnomeontherun commented on src/clarity-angular/utils/_theme.dark.clarity.scss:
```diff
@@ -0,0 +1,694 @@
+ * - ../image/_icons.clarity.scss (login background image)
+ */
+$clr-login-background-color: $clr-global-app-background;
+$clr-login-background: "%3C%3Fxml%20version%3D%221.0%22%20encoding%3D%22UTF-8%22%20standalone%3D%22no%22%3F%3E%3Csvg%20width%3D%22736px%22%20height%3D%22838px%22%20viewBox%3D%220%200%20736%20838%22%20version%3D%221.1%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20xmlns%3Axlink%3D%22http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%22%3E%3Ctitle%3Evector%20art%3C%2Ftitle%3E%3Cdesc%3ECreated%20with ...
```
I don't like hosting SVG data directly in our CSS files, especially for just one of the pages (login). I see we do this already though :(

hippee-lee replied:
I believe most of these were added before `Clarity Icons` was ready. So this change took the original code and added to it so the consumers couldoverride the bg image of the login form. 

We have talked about a larger initiative/issue to remove svg images from, our css.

gnomeontherun commented on src/clarity-angular/utils/_theme.dark.clarity.scss:
```diff
@@ -0,0 +1,694 @@
+ * Usage: ../progress/progress-bars/_progress-bars./clarity.scss
+ */
+$clr-progress-defaultBarColor: #49afd9;
+$clr-progress-success-color: #205522; // TODO confirm these with UX - reused existing success/danger colors from elsewhere
```
Marking this as something to ensure is ok?

hippee-lee replied:
I did confirm once and will remove the comment.

gnomeontherun commented on webpack.config.js:
```diff
@@ -374,6 +422,8 @@ module.exports = {
       new SuppressChunksPlugin([
           { name: 'clarity-ui/clarity-ui', match: /\.js(\.map)?$/ },
           { name: 'clarity-ui/clarity-ui.min', match: /\.js(\.map)?$/ },
+          { name: 'clarity-ui/clarity-ui-dark', match: /\.js(\.map)?$/ },
```
Is this going to add another file to the dev server that is 404?

hippee-lee replied:
Yes.

gnomeontherun commented on src/ks-app/.angular-cli.json:
```diff
@@ -20,7 +20,7 @@
       "styles": [
         "../node_modules/clarity-icons/clarity-icons.min.css",
-        "../node_modules/clarity-ui/clarity-ui.min.css",
+        "../node_modules/clarity-ui/clarity-ui-dark.min.css",
```
Is it possible to flip back to before dark theme for KS easily or have you removed things that make it look alright?

hippee-lee replied:
I touched on this in the overview last week ... its one of the less than optimal dev items that I need to figure out. With how we are using webpack the best way to `flip` back and forth is to comment/uncomment the dark theme variables file while running the demo app in `_theme.dark.clarity.scss`

One idea I had is to split them out and export multiple configurations(https://webpack.js.org/configuration/configuration-types/#exporting-multiple-configurations) so that both css files get added into the demo app and we can add a component into the app that toggles between the two stylesheets.

hippee-lee commented:
@lil-kim 
Updates for the following have been published here: http://clarity-ks-dark.surge.sh/alerts

- Checkbox and radio colors (disabled states)
- Removed clickable labels for label types in ks-app
- Updated signpost (and all popovers in datagrid) background color
- Updated modal overlay background-color
- Updated progress bar success and danger colors and the background-color for the progress track

lil-kim commented:
Vertical nav section  - Button text color in example is wrong.
![image](https://user-images.githubusercontent.com/23196444/32866512-1029236e-ca1d-11e7-8ad9-3d7a3186657e.png)

lil-kim commented:
Hi @hippee-lee,
Can we apply a Dark theme header color, #404E60? By "we", I mean you. 😄 

![image](https://user-images.githubusercontent.com/23196444/32866718-f98962d0-ca1d-11e7-9495-36540907dce1.png)

mathisscott commented on src/clarity-angular/utils/_variables.global.scss:
```diff
@@ -0,0 +1,79 @@
+// App hover color
+$clr-global-hover-color: $gray-light;
+// App border radius
+$clr-global-borderradius: $clr_baselineRem_0_125 !default;
```
`$clr-global-borderradius` and `$clr-global-borderwidth` should refer to `$clr-default-borderwidth` and `$clr-default-borderradius` (or vice versa) because the `global` variables are intended to replace `default`.

We want to deprecate the `default` ones for the new naming convention. But what we have done instead is provide an end-around that allows users to customize one and not the other.

hippee-lee replied:
I made `$clr-global-*` refer to the `$clr-default` so that it will be easier to remove the defaults in the future.

mathisscott commented on src/clarity-angular/utils/_variables.global.scss:
```diff
@@ -0,0 +1,79 @@
+// App hover color
+$clr-global-hover-color: $gray-light;
+// App border radius
+$clr-global-borderradius: $clr_baselineRem_0_125 !default;
```
We shouldn't be using `$clr_baselineRem_0_125` here or anywhere. The `$clr_baselineRem_*` utility classes are deprecated and no longer necessary.

Use `0.125rem` instead.

hippee-lee replied:
I changes this and also grouped them with the deprecated variables that are used to set the new $clr-global variable.

mathisscott commented on src/clarity-angular/utils/_variables.global.scss:
```diff
@@ -0,0 +1,79 @@
+$clr_baselineRem_3_5: baselineRem(3.5);
+$clr_baselineRem_4: baselineRem(4);
+$clr_baselineRem_5: baselineRem(5);
+$clr_baselineRem_6: baselineRem(6);
```
All `$clr_baselineRem_*` utlities are deprecated. None of them should be calling the `baselineRem()` function — which is also deprecated.

Use regular `rem` values instead.

So...

`$clr_baselineRem_0_1: 0.1rem;`

..instead of...

`$clr_baselineRem_0_1: baselineRem(0.1);`

hippee-lee replied:
Will update.

hippee-lee replied:
Updated and made a note about deprecation.

mathisscott commented on src/clarity-angular/utils/_variables.global.scss:
```diff
@@ -0,0 +1,79 @@
+// App bg color
+$clr-global-app-background: $gray-lighter;
+// App selection color
+$clr-global-selection-color: #D9E4EA;
```
We need to add...

`$clr-selection-color: $clr-global-selection-color;`

...here. Otherwise, the removal of `$clr-selection-color` will be a breaking change.

Please add a comment that `$clr-selection-color` is deprecated.

hippee-lee replied:
I will add this.

Shijir commented on src/clarity-angular/wizard/_variables.wizard.scss:
```diff
@@ -0,0 +1,42 @@
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+$clr-wizard-main-bgcolor: clr-getColor(1);
```
These default variables don't have `!default` keyword in their values so users wouldn't be able to overwrite them if they want to build on their own.

hippee-lee replied:
Most of the variables were migrated `as is` from the original location.
I do override several of the wizard colors for dark theme so this won't prevent users from overwriting them in a custom theme.

Shijir commented on src/clarity-angular/progress/progress-bars/_variables.progress-bars.scss:
```diff
@@ -0,0 +1,12 @@
+ */
+
+// Usage: ../progress/progress-bars/_progress-bars.clarity.scss
+$clr-progress-defaultBarColor: $clr-action-blue;
```
`!default` keyword is from these default variables?

hippee-lee replied:
Most of the variables were migrated `as is` from the original location.
I do override several of the wizard colors for dark theme so this won't prevent users from overwriting them in a custom theme.

Shijir commented on src/clarity-angular/utils/_variables.clarity.scss:
```diff
@@ -2,508 +2,187 @@
+ * Typography
+ */
+@import "../typography/variables.typography";
+// End Typography
```
Awesome refactoring 👍

hippee-lee replied:
Thanks :-)

hippee-lee commented on src/clarity-angular/wizard/_variables.wizard.scss:
```diff
@@ -0,0 +1,42 @@
+$clr-wizard-stepnav-border-color: clr-getColor(3);
+$clr-wizard-stepnav-border-color--active: $clr-green;
+$clr-wizard-footer-height: 3.5rem;
+$clr-wizard-stepnav-active-bgcolor: #D9E4EA;
```
Most of the variables were migrated `as is` from the original location.
You bring up a good point though. One thing I ran into is that coming up with semantic color names that can apply to fonts, backgrounds, borders indiscriminately is hard and not all of the colors are in our color palettes. I'll leave a note so we remember to address it when we focus on theme colors.

hippee-lee commented on webpack.config.js:
```diff
@@ -177,7 +211,9 @@ module.exports = {
         "exclude": [
+
           path.join(process.cwd(), "src/clarity-angular/main.scss"),
+          // path.join(process.cwd(), "src/clarity-angular/dark-theme.scss"),
```
I'll change it to keep it uncommented. We need it when we have to develop or debug a component with the dark theme applied.
